lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -573,51 +573,30 @@ Sample DistributionImplementation::getSample(const UnsignedInteger size) const
 /* Get one realization of the distribution */
 Point DistributionImplementation::getRealizationByInversion() const
 {
-  // Use CDF inversion in the 1D case
-  if (dimension_ == 1) return computeQuantile(RandomGenerator::Generate());
-  // Use conditional CDF inversion in the 1D case
-  Point point(0);
-  for (UnsignedInteger i = 0; i < dimension_; ++ i)
-  {
-    const Scalar u = RandomGenerator::Generate();
-    LOGINFO(OSS(false) << "i=" << i << ", u=" << u);
-    point.add(computeConditionalQuantile(u, point));
-    LOGINFO(OSS(false) << "i=" << i << ", u=" << u << ", point=" << point);
-  }
-  return point;
+  return getSampleByInversion(1)[0];
 }
 
 /* Get a numerical sample whose elements follow the distributionImplementation */
 Sample DistributionImplementation::getSampleByInversion(const UnsignedInteger size) const
 {
   // Use CDF inversion in the 1D case
   if (dimension_ == 1) return computeQuantile(RandomGenerator::Generate(size));
-  // Use conditional CDF inversion in the 1D case
-  Sample result(size, 0);
-  for (UnsignedInteger i = 0; i < dimension_; ++ i)
-  {
-    const Point u(RandomGenerator::Generate(size));
-    SampleImplementation q(size, 1);
-    q.setData(computeConditionalQuantile(u, result));
-    result.stack(q);
-  }
+  // Use conditional CDF inversion in the nD case
+  Sample result(size, dimension_);
+  for (UnsignedInteger i = 0; i < size; ++ i)
+    result[i] = computeSequentialConditionalQuantile(RandomGenerator::Generate(size));
```

Comment:
for n-d, it segfaults in computeSequentialConditionalQuantile at line 2237 because the loop goes up to dimension_ instead of the quantile size

Reply:
You mean, `RandomGenerator::Generate(size)` should be `RandomGenerator::Generate(dimension_)`? True, my bad!